src/hooks/useStakingH160.ts

```diff
@@ -0,0 +1,204 @@
+    async () => {
+      if ($web3.value && addressRef.value) {
+        const ci = contractInstance(
+          // $web3.value,
+          getDefaultEthProvider(),
+          dappsStakingContractAbi,
+          PRECOMPILED_ADDR,
+          addressRef.value
+        );
+        stakingRef.value = new Staking(ci, addressRef.value);
+      }
+    },
+    { immediate: true }
+  );
+
+  const getEraInfo = () => {
+    const result = ref();
+    watch(
+      () => stakingRef.value,
+      async () => {
+        if (stakingRef.value) {
+          result.value = {
+            currentEra: await stakingRef.value.getCurrentEra(),
+            unbondingPeriod: await stakingRef.value.getUnbondingPeriod(),
+          };
+        }
+      },
+      { immediate: true }
+    );
+
+    return { result };
+  };
+
+  const getEraRewardsAndStakes = (era: number) => {
+    const result = ref();
+    watch(
+      () => stakingRef.value,
+      async () => {
+        if (stakingRef.value) {
+          result.value = {
+            eraReward: await stakingRef.value.getEraReward(era),
+            eraStaked: await stakingRef.value.getEraStaked(era),
+          };
+        }
+      },
+      { immediate: true }
+    );
+
+    return { result };
+  };
+
+  const getStakedAmount = (staker: string) => {
+    const result = ref();
+    watch(
+      () => stakingRef.value,
+      async () => {
+        if (stakingRef.value) {
+          result.value = {
+            stakedAmount: await stakingRef.value.getStakedAmount(staker),
+          };
+        }
+      },
+      { immediate: true }
+    );
+
+    return { result };
+  };
+
+  const getContractStake = (contract_id: string) => {
+    const result = ref();
+    watch(
+      () => stakingRef.value,
+      async () => {
+        if (stakingRef.value) {
+          result.value = {
+            stakedAmount: await stakingRef.value.getContractStake(contract_id),
+          };
+        }
+      },
+      { immediate: true }
+    );
+
+    return { result };
+  };
+
+  /* extrinsic calls */
+  const handleError = (e: any) => {
+    console.error(e);
+    store.dispatch('general/showAlertMsg', {
+      msg: `Transaction failed with error: ${e}`,
+      alertType: 'error',
+    });
+  };
+
+  const callRegister = async (contractAddr: string) => {
+    let txHash;
+    store.commit('general/setLoading', true);
+    try {
+      txHash = await stakingRef.value.callRegister(contractAddr);
+      console.log('ss', txHash);
+      store.dispatch('general/showAlertMsg', {
+        msg: `transaction is confirmed: ${txHash}`,
+        alertType: 'success',
+      });
+    } catch (e) {
+      handleError(e);
+    }
+    store.commit('general/setLoading', false);
+    return txHash;
+  };
+
+  const callBondAndStake = async (contractAddr: string, amount: BN) => {
+    let txHash;
+    store.commit('general/setLoading', true);
+    try {
+      txHash = await stakingRef.value.callBondAndStake(contractAddr, amount);
```

Comment:
![Screen Shot 2022-03-04 at 5 15 43 PM](https://user-images.githubusercontent.com/40356749/156799447-a10deb45-8d07-41de-b6f6-5ae27dc3b16a.png)
When I try to stake, it always says the sending amount is 0. Is this normal?

Reply:
yes, I also thought I should send the value too, but when I check the amount I staked and the reduced balance, it seems to be working well. 
@bobo-k2 can you please double-checking on your end?

Reply:
@sirius651 @hoonsubin 
Is it possible to modify the contract side to make `bond_and_stake` function [payable](https://github.com/AstarNetwork/astar-apps/blob/626cc28eb97e2d45b26a820b51238d6a2eab4ad7/src/config/web3/abi/dapps-staking-abi.json#L17) instead of taking the 2nd parameter(amount)?  Then we can add `value: amount` in [send](https://github.com/AstarNetwork/astar-apps/pull/257/files#diff-25c11cfa13bc80ad483585a408389b8eefa807d56041f9b063ac943887a301aaR60) function.

![Screenshot 2022-03-07 at 9 54 49 PM](https://user-images.githubusercontent.com/92044428/157049238-9f236070-b166-47f6-8efa-d41a2b086b37.png)

Reply:
yeah, that's more makes sense. just not sure how much it will affect the core contract though

Reply:
Hi @Maar-io
What do you think about [this](https://github.com/AstarNetwork/astar-apps/pull/257#discussion_r820736176) comment? :)

Reply:
The staking is not payable function. When you stake on ds there is no sending funds. Funds are just locked